flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialectFactory.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.jdbc.dialect;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+/**
+ * A factory to create different JdbcDialect's. This factory is used with Java's Service Provider
+ * Interfaces (SPI) for discovering.
+ *
+ * <p>Classes that implement this interface can be added to the
+ * "META_INF/services/org.apache.flink.connector.jdbc.dialect.JdbcDialectFactory" file of a JAR file
+ * in the current classpath to be found.
+ *
+ * @see JdbcDialect
+ */
+@PublicEvolving
+public interface JdbcDialectFactory {
```

Comment:
What's the benefit of going through the factory over loading the `JdbcDialect` instances directly via the ServiceLoader?

Reply:
Mostly it's just that a factory is in line with how the rest of the project deals with service loaders. I agree it doesn't add much in this particular situation but it also doesn't hurt and I prefer the consistency.

Reply:
The factory would allow an implementation a bit more freedom if constructing the class isn't that simple. Probably not a realistic use case, but I agree that it doesn't hurt to design the API in an extensible way.